proto-public/pbhcp/v1/link.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package hashicorp.consul.hcp.v1;
```

Comment:
we decided to add all the new resource api as v2. To reduce the confusion when discussion the old api vs the new one. Maybe you want to consider doing the same with hcp?

Reply:
I think there was some desire to keep this as distinct from the "v2 resources". For example, this feature should be able to be enabled independently of whether "v2 resources" are enabled (in fact, for now if "v2 resources" is enabled, this `link` resource will not work actually). In that sense, does it make it more confusing to lump them together? I'll defer to you on this Dhia, I just wanted to raise these points and see what you think.

TBH I'm a little unclear on what the versioning strategy is for these resources — should any new resource created now start at `v2`? Or should this be `v1` since there was not previously another version of the `Link` resource?

Reply:
This is still a resource, so I don't see how we would make it distinct for v2 resources.

Resources are independently versioned, so from functional perspective versioning this v1 is ok. That said, I feel it's weird to have all resources starting a v2 (for now v2beta1 and will change to v2 when GA'd) except this resource, so maybe we can decide that v2 is our first version of resources and skip v1 all together.

In the future all resource types versions will change independently based on changes and compatibility concerns.

Reply:
Done